c7n/resources/asg.py

```diff
@@ -820,6 +822,49 @@ def process(self, asgs, event=None):
                 len(a['Instances']) < a['MinSize']]
 
 
+@filters.register('user-data')
+class UserDataFilter(ValueFilter, LaunchConfigFilterBase):
+
+    schema = type_schema('user-data', rinherit=ValueFilter.schema)
+    batch_size = 50
+    annotation = 'c7n:user-data'
+
+    def validate(self):
```

Comment:
this method can be omitted, since its not doing any validation, to inherit the base class.

Reply:
Should it be taken out of ec2.py as well?

Reply:
its just returning self, sure.